python/cudf/cudf/core/column/column.py

```diff
@@ -135,6 +135,16 @@ def to_pandas(self):
             sr[~mask_bytes] = None
         return sr
 
+    def clip(self, lo, hi):
+        # If categorical, it returns non categorical column
+        # as there is possibility of change in categories
+        if is_categorical_dtype(self):
+            input_col = self.astype(self.categories.dtype)
```

Comment:
We don't have to follow Pandas behavior in this regard. We can make clip return the same CategoricalDtype but clip the values as expected.